Automattic/wp-calypso
# Expired migration: can not delete site because of 'paid' plan

This PR fixes two bugs:

1. A WoA site can't be deleted. This PR removes the "Delete site" buttons from expired pages' fullscreen trail
2. Trial plans are not "purchases," so they must not appear in the site purchases list.

Fixes https://github.com/Automattic/dotcom-forge/issues/3556

## Proposed Changes

* Remove the buttons
* Make check for trial sites on purchases list

## Testing Instructions



* Get an expired WooCommerce and a business plan site
* Open `http://calypso.localhost:3000/plans/my-plan/trial-expired/__BLOG__` or simply try to open other pages, you will be redirected
* Check that the delete buttons have been removed
* Run new unit tests `yarn run test-client client/state/selectors/test/has-cancelable-site-purchases.js`

## Screenshot

![screenshot](https://github.com/Automattic/wp-calypso/assets/167611/0cca6584-98a0-4009-94fd-025fe1e3165e)

<img width="740" alt="image" src="https://github.com/Automattic/wp-calypso/assets/167611/27f526b5-5d3b-4ffd-bd85-6e64e2351284">

## Pre-merge Checklist
